### PR TITLE
Include path to the original XSpec test in all output report formats

### PR DESCRIPTION
**Pull request by vincentml**
This pull request adds a reference to the original XSpec file in all output report formats and is consistent for XSLT, XQuery, and Schematron tests.

The original XSpec file location is included in the XSpec XML report as attribute `x:report/@xspec-original-location`; in the JUnit report as attribute `testsuites/@name`; and in the HTML report as part of test report summary information. 

In Schematron tests the original XSpec file location is added to the temporary XSpec file as attribute `x:description/@xspec-original-location`.

This pull request also improves the HTML report summary information for Schematron tests by including the path to the Schematron being tested in a way consistent with XSLT and XQuery test reports, and removed the generated scenario with Schematron information in a label.

The end-to-end tests have also been extended to include XQuery and Schematron tests.

This pull request closes issue #176, issue #180, and pull request #181 .

### CODE REVIEW

**Comment by cmarchand** [comment]
This pull request if Ok for me, and I already use this code in `xspec-maven-plugin`.

File: test/end-to-end/processor/_normalizer.xsl [hunk 1]
```diff
@@ -81,7 +81,7 @@
 				in:		<a href="file:/path/to/tested.xsl">/path/to/tested.xsl</a>
 				out:	<a href="tested.xsl">tested.xsl</a>
 	-->
-	<xsl:template as="element(a)" match="/html/body/p[1]/a" mode="local:normalize">
+	<xsl:template as="element(a)" match="/html/body/p[1 or 2]/a" mode="local:normalize">
```
Comment (AirQuick) [inline test/end-to-end/processor/_normalizer.xsl]: Meant to be `[position() = (1, 2)]`?

**Comment by AirQuick** [comment]
Nit picking.
Looking at a report XML, 
```xml
<x:report ...
          stylesheet=".../demo-01.sch-compiled.xsl"
          xspec-original-location=".../demo-01.xspec"
          schematron=".../demo-01.sch">
```
I wonder why the attribute name couldn't be just `@xspec` which seems better fit with the other attributes?

File: test/end-to-end/processor/_normalizer.xsl (hunk 1 above)
Comment (vincentml) [reply test/end-to-end/processor/_normalizer.xsl]: Both predicates work. I've changed this to the more explicit version as suggested.

**Comment by vincentml** [comment]
I've changed the name of the attribute that holds the XSpec file location in the XML report to `x:report/@xspec` as suggested by @AirQuick. 

In the temporary XSpec file that is generated for Schematron tests the XSpec file location is still in attribute `x:description/@xspec-original-location`.

**Comment by vincentml** [comment]
Once this pull request has been merged I will create a pull request to extend the end-to-end test to include JUnit report output. This secondary change is on branch test/e2e-junit.

**Comment by AirQuick** [comment]
> Once this pull request has been merged I will create a pull request to extend the end-to-end test to include JUnit report output. This secondary change is on branch test/e2e-junit.

Cool.

Can you possibly review #187 later on?
As written there, I have a not-yet-submitted pull request to update end-to-end. JUnit processor is missing from it.
So I'm thinking we could align in this sequence:

1. Merge this (#188) with my review.
1. You submit `test/e2e-junit`. Merge it with my review.
1. Merge #187 with your review.
1. I submit a pull request: End-to-end test using #187-based driver and `test/e2e-junit`-based processors

What do you think?

**Comment by vincentml** [comment]
@AirQuick That sounds like a good plan!

**Comment by AirQuick** [comment]
cirulls seems so busy that he gave consent to my coordinated facilitation of pull requests.
I'll merge this shortly.

**Comment by cmarchand** [comment]
Great !
I'll need a release as soon as possible, to be able to release xspec-maven-plugin with all these features. A 1.0.1, as there is no backward changes ?

**Comment by AirQuick** [comment]
@cirulls 
Merged this.

**Comment by AirQuick** [comment]
@cmarchand 
The release will be *roughly* tracked [here](https://github.com/xspec/xspec/projects/2).

**Pull request merged**